### PR TITLE
Alien Cuisines

### PR DESCRIPTION
**Pull request by VernaSaurus**
Hey, you! Yeah you, that cook over there. Have you ever wanted to make some species-specific dishes for your moth and lizard friends?

No?

Just me?

Well now you can do that anyway! With over a hundred new foods of all different kinds, and tons of drinks for the bar, you can finally get a taste of home- if you're a moth or a lizard, that is.

### CHANGELOG:
CL: [New] added like a bajillion foods and drinks and all their recipes
CL: [Improvement] refactors the grinder so it isn't shit and adds like fifty recipes so you can grind or juice to your heart's content
CL: [New] added drying racks which you can make with wood planks and are used to cure headcheese and blood sausages and make firm cheese
-removed about a year off of my lifespan

All of the sprites are taken from /tg/.
https://github.com/tgstation/tgstation/pull/61743
https://github.com/tgstation/tgstation/pull/59412

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/Kitchen/Dryable.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	[Tooltip("Minimum time to dry.")]
+	public int DryTime = 10;
+
+	[InfoBox("If no transormation is to take place, then don't select any item. Don't select the same item as itself.", EInfoBoxType.Warning)]
+	[Tooltip("What this item becomes when dried." +
+				"If not set, this item will not change GameObject when dried, but will still invoke the dried event.")]
+	public GameObject DriedProduct;
+
+	/// <summary>
+	/// Raised when enough cooking time has been added (via <see cref="AddCookingTime(float)"/>)
```
Comment (pyrosshade) [inline UnityProject/Assets/Scripts/Items/Kitchen/Dryable.cs]: Probably want to adjust `(via <see cref="AddCookingTime(float)"/>)` to be 
`(via <see cref="AddDryingTime(float)"/>)`

**Comment by corp-0** [comment]
why does it change so many seemingly unrelated files? Like metal rods

File: UnityProject/Assets/Scripts/Items/Kitchen/Dryable.cs (hunk 1 above)
Comment (VernaSaurus) [reply UnityProject/Assets/Scripts/Items/Kitchen/Dryable.cs]: shit you're right

**Comment by VernaSaurus** [comment]
> why does it change so many seemingly unrelated files? Like metal rods

crafting recipes. lots of crafting recipes. also a bunch of things have the grindable component added.

**Review by Bod9001 — DISMISSED** [review]

File: UnityProject/Assets/Scripts/Items/Kitchen/Dryable.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+	[InfoBox("If no transormation is to take place, then don't select any item. Don't select the same item as itself.", EInfoBoxType.Warning)]
+	[Tooltip("What this item becomes when dried." +
+				"If not set, this item will not change GameObject when dried, but will still invoke the dried event.")]
+	public GameObject DriedProduct;
+
+	/// <summary>
+	/// Raised when enough drying time has been added (via <see cref="AddDryingTime(float)"/>)
+	/// </summary>
+	public event Action OnDried;
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Items/Kitchen/Dryable.cs:19]: Is this used on prefabs?

File: UnityProject/Assets/Scripts/Objects/Kitchen/InteractableDryingRack.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	/// Allows food processor to be interacted with. Player can put food in the processor to process it.
+	/// The processor can be toggled on and off, eject contents, or have something Processable put inside it.
+	/// </summary>
+	[RequireComponent(typeof(DryingRack))]
+	public class InteractableDryingRack : MonoBehaviour, IExaminable, ICheckedInteractable<HandApply>,
+			IRightClickable, ICheckedInteractable<ContextMenuApply>
+	{
+		private DryingRack dryingRack;
+
+		private void Start()
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Objects/Kitchen/InteractableDryingRack.cs]:
```suggestion
		private void Awake()
```

File: UnityProject/Assets/Scripts/Objects/Machines/DryingRack.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+			UpdateManager.Remove(CallbackType.UPDATE, UpdateMe);
+		}
+
+		/// <summary>
+		/// Count down processor timer.
+		/// </summary>
+		private void UpdateMe()
+		{
+			CheckDried(Time.deltaTime);
+		}
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Objects/Machines/DryingRack.cs]: Not sure i like this running every frame, maybe every 0.5 seconds would be better using the periodic update?

**Review by ThatDan123 — CHANGES_REQUESTED** [review]

File: UnityProject/Assets/Scripts/Objects/Machines/DryingRack.cs (hunk 4 above)
Comment (VernaSaurus) [reply UnityProject/Assets/Scripts/Objects/Machines/DryingRack.cs]: Sounds good to me.

File: UnityProject/Assets/Scripts/Items/Kitchen/Dryable.cs (hunk 2 above)
Comment (VernaSaurus) [reply UnityProject/Assets/Scripts/Items/Kitchen/Dryable.cs:19]: Yes, it's used on Cheese Curds, Raw Headcheese, and Raw Blood Sausage.

**Review by ThatDan123 — APPROVED** [review]

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**